app/models/nextUpdates/NextUpdatesResponseModel.scala

```diff
@@ -66,6 +67,30 @@ case class ObligationsModel(obligations: Seq[NextUpdatesModel]) extends NextUpda
 
   def obligationsByDate(implicit mtdItUser: MtdItUser[_]): Seq[(LocalDate, Seq[NextUpdateModelWithIncomeType])] =
     allDeadlinesWithSource().groupBy(_.obligation.due).toList.sortWith((x, y) => x._1.isBefore(y._1))
+
+  def getPeriodForQuarterly(obligation: NextUpdateModelWithIncomeType): String = {
```

Comment:
Looks good, just one more extra step as per this reference PR: https://github.com/hmrc/income-tax-view-change-frontend/pull/2081/files (please feel free to continue with the PR you created):
* rename Standard and Calendar as its not clear what these are + change to case object, not sure why we design these as a singleton in the first place;
* change groupByQuarterPeriod method to return Map[Option[QuarterReportingType], Seq[NextUpdateModelWithIncomeType]] or Map[QuarterReportingType, Seq[NextUpdateModelWithIncomeType]]  ? have a look at the reference PR;